tutorons/python/explain.py

```diff
@@ -0,0 +1,44 @@
+#! /usr/bin/env python
+# encoding: utf-8
+
+from __future__ import unicode_literals
+import logging
+from bs4 import BeautifulSoup
+import collections
+
+logging.basicConfig(level=logging.INFO, format="%(message)s")
+
+fcns = open('tutorons/python/fcns.txt', 'r').read()
+builtin_doc = BeautifulSoup(fcns)
+dt = builtin_doc.findAll('dt')  # headers
+dd = builtin_doc.findAll('dd')  # explanations
+explanations = collections.OrderedDict()  # builtin : explanation (dt, dd)
+
+
+def resolve(a):
+    """Resolve relative hyperlinks to poin to python docs instead of current page"""
+    a['href'] = a['href'].replace(a['href'], 'https://docs.python.org/2/library/' + a['href'])
+
+# fix all relative hyperlinks
+for tag in dd:
+    map(resolve, tag.findAll('a'))
+
+# group dt tags for the same builtin and add to explanations dict
+for tag in dt:
+    if tag.code.text.encode() not in explanations:
+        explanations[tag.code.text.encode()] = tag
```

Comment:
Take a look [here](http://www.w3.org/TR/html5/grouping-content.html#the-dl-element) for a discussion of the expected structure and semantics of `dl`, `dd`, `dt`.  Looks like typical usage in markup is, if you have multiple `dt` tags for a term-definition pair, they should be sequential elements---one after the other---rather than nested.  Can we aim for this more typical usage of `dt` in how we add it to the template?